samples/wifi/softap/src/wpas_events.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/net/net_event.h>
+#include <ctrl_iface_zephyr.h>
+#include <supp_events.h>
+#include <supp_main.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+LOG_MODULE_REGISTER(wpas_event, CONFIG_LOG_DEFAULT_LEVEL);
+
+/* TODO: Handle other events */
+#define WPA_SUPP_EVENTS (NET_EVENT_WPA_SUPP_READY)
+
+static struct net_mgmt_event_callback net_wpa_supp_cb;
+
+K_SEM_DEFINE(wpa_supp_ready_sem, 0, 1);
+
+static void handle_wpa_supp_ready(struct net_mgmt_event_callback *cb)
```

Comment:
Rather copy-pasting this file in each and every sample, should we just make a library or a common helper to avoid duplication?

Reply:
Planning to take up this as another task. Will handle this as a separate task.

Reply:
Well, that would mean that another PR will just revert all these changes? I would suggest fixing this sample in another PR with a simple sleep + TODO comment and then use this PR to properly implement WPA supplicant events for all samples. WDYT?